cumulus/steps/ec2/ingress_rule.py

```diff
@@ -0,0 +1,33 @@
+from troposphere import elasticloadbalancingv2 as alb
+from troposphere import (
+    Ref, ec2)
+import re
+from cumulus.chain import step
+from cumulus.steps.ec2 import META_SECURITY_GROUP_NAME
+
+
+class IngressRule(step.Step):
+
+    def __init__(self,
+                 port_to_open,
+                 cidr):
+
+        step.Step.__init__(self)
+
+        self.port_to_open = port_to_open
+        self.cidr = cidr
+
+    def handle(self, chain_context):
+        template = chain_context.template
+
+        clean_cidr = re.compile('[\W_]+').sub('', self.cidr)
+
+        template.add_resource(ec2.SecurityGroupIngress(
+            "Cidr%sToASGPort%s" % (clean_cidr, self.port_to_open),
+            IpProtocol="tcp",
+            FromPort=self.port_to_open,
+            ToPort=self.port_to_open,
+            CidrIp=self.cidr,
+            GroupId=Ref(chain_context.metadata[META_SECURITY_GROUP_NAME])
```

Comment:
This is a slight departure from a convention the pipeline introduced..  it's not a big deal but might be good to streamline as much as possible. 
Instead of popping the name on the metadata, pipeline put the Ref object on it.   Which makes the most sense?  Should the pipeline change or should this?

Reply:
I found putting it on the metadata was pretty intuitive!

Reply:
so did I.   The question was should we put a string or a ref.    If we stick with the precedence in master we'd put a Ref object on the metadata.